connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/DefaultProcessInstanceContext.java

```diff
@@ -68,7 +65,8 @@ public Long getKey() {
 
   @Override
   public <T> T bind(final Class<T> cls) {
-    HashMap<String, Object> copyOfProperties = new HashMap<>(contextProperties);
+    // TODO we should improve caching to avoid fetching variables for very bind.
```

Comment:
Shall we also have a ticket for the `TODO`?

Reply:
@igpetrov Yes, added reference ticket